server/service/device_client.go

```diff
@@ -50,11 +73,16 @@ func NewDeviceClient(addr, token string, insecureSkipVerify bool, rootCA string)
 
 // ListDevicePolicies fetches all policies for the device with the provided token
 func (dc *DeviceClient) ListDevicePolicies() ([]*fleet.HostPolicy, error) {
-	verb, path := "GET", "/api/latest/fleet/device/"+dc.token+"/policies"
+	verb, path := "GET", "/api/latest/fleet/device/"+dc.GetToken()+"/policies"
 	var responseBody listDevicePoliciesResponse
 	err := dc.request(verb, path, "", &responseBody)
 	if err != nil {
 		return nil, err
 	}
 	return responseBody.Policies, nil
 }
+
+func (dc *DeviceClient) Check(token string) error {
```

Comment:
Uhmm, what are your thoughts about passing in the token to every method in `DeviceClient`? It might be a matter of coding style, but for me, that signals that the token might change between calls.

Reply:
I was in the fence about this when I wrote it, since you brought it up seems like we have one and a half vote vs half 😅 , I changed it 👍